core/src/Cuda/Kokkos_Cuda_Parallel_MDRange.hpp

```diff
@@ -63,9 +63,19 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
     auto const& prop = pol.space().cuda_device_prop();
     // Limits due to registers/SM, MDRange doesn't have
     // shared memory constraints
-    int const regs_per_sm        = prop.regsPerMultiprocessor;
-    int const regs_per_thread    = attr.numRegs;
-    int const max_threads_per_sm = regs_per_sm / regs_per_thread;
+    auto const block_size_to_zero_shmem = [&] (int block_size) {
+      return 0;
+    };
+    int const optimal_block_size = Kokkos::Impl::cuda_deduce_block_size(false, prop, attr, block_size_to_zero_shmem, LaunchBounds{});
```

Comment:
This is a bit awkward.  This function was never intended to be exposed beyond the `<Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp>` header.  Maybe we could define an `Impl::cuda_get_opt_block_size_no_shmem()` that does not have the leading boolean parameter nor the lambda.

Reply:
Sure, I can do that today/early tomorrow. Good suggestion.

Reply:
Added said new function.